lib/DxilDia/DxilDiaDataSource.cpp

```diff
@@ -61,9 +61,9 @@ getMemBufferFromStream(IStream *pStream, const llvm::Twine &BufferName) {
   size_t size = statstg.cbSize.LowPart;
   std::unique_ptr<llvm::MemoryBuffer> result(
       llvm::MemoryBuffer::getNewUninitMemBuffer(size, BufferName));
```

Comment:
Rather than create this uninitialized, we could create an allocation of some kind to read the stream into and then create the membuffer using a stringref to that allocation.  That would prevent the need to cast the const off on line 66.